### PR TITLE
rework pencils error logic [wip]

### PR DESCRIPTION
**Pull request by robx**
From my point of view this is done now, but I feel I may well have missed things.

Would be happy for your review.

### CODE REVIEW

**Comment by robx** [comment]
It's live now at https://puzz.link/pztest/p.html?pencils/10/10/lim6q1lgk2l4xim4ljmjp9rgki2v3hkilip3p (link to a pencils puzzle off twitter)

File: src/variety/pencils.js
Comment (x-sheep) [inline src/variety/pencils.js:586]: Could you make sure these checks are at the top of the list?

* Lines cross or branch off
* A line crosses a pencil
* A line doesn't stop at a pencil tip (lcnt > 1)
* A line connects to two pencil tips
* A line is too long for this pencil
* A number is too large for this pencil

**Review by x-sheep — CHANGES_REQUESTED** [review]
Hm, turns out my notification settings aren't right either 😥 Sorry for the late reply.

The 5kB difference in the source length is definitely a good reason to overhaul the error model.

My first impression is that the Check button feels fairly useless in this version. Obvious mistakes are only pointed out when the grid is nearly full. I prefer having some of the checks split up from `expected !== actual` into two different checks for higher and lower values, where the mistake is checked first. This is in line with the implementations for Creek, Amibo, Fillomino and many other puzzles.

**Comment by robx** [comment]
No worries, I wasn't blocked on your response, just got side-tracked. But I'm happy for your input!

Just to be sure I understand, you want the "higher" check first because that's clearly wrong, while the "lower" check might be on the path to a correct solution, right?

I also still want to put in a different tip input mode before merging, cycling through a cell by clicking it, at the expense of background-shading in auto-mode.

Double Chocolate has been taken up very well by the way, thanks again for that one! http://puzsq.sakura.ne.jp/main/index.php?puzzle=66&author=0&page=5

File: src/variety/pencils.js
Comment (x-sheep) [reply src/variety/pencils.js:586]: I've implemented these changes in https://github.com/x-sheep/pzprjs/tree/pr-pencil. If it looks good to you, I'll add them to [my pending PR](https://github.com/robx/pzprjs/pull/40) and make it up-to-date with `master`.

Comment (robx) [reply src/variety/pencils.js:586]: Yes please do! I'd be happy to get this merged soon. I should have time to merge this and make the pencil tip input change later this week. Sorry for the wait.

**Pull request merged**
